### PR TITLE
add swin transformer

### PR DESCRIPTION
**Pull request by littletomatodonkey**
att, four models top1 accs on ImageNet1k are as follows.

```
base_patch4_window12_384: {'top1_acc': 0.84390}
base_patch4_window7_224: {'top1_acc': 0.8299600000762939}
small_patch4_window7_224: {'top1_acc': 0.82746}
tiny_patch4_window7_224: {'top1_acc': 0.806879999961853}
```

* todo
    * support training

* Current problem
    * If you train the transformer using GPU and encountered the following problem, you can set `arch=sm_61`, more detailed info can be seen in [link](https://github.com/vchoutas/torch-mesh-isect/issues/6)

```
ERROR - (Fatal) Operator uniform_random raises an thrust::system::system_error exception.
The exception content is
:parallel_for failed: cudaErrorInvalidConfiguration: invalid configuration argument. (at /paddle/paddle/fluid/imperative/tracer.cc:172)
```

### CODE REVIEW

File: ppcls/modeling/architectures/swin_transformer.py [hunk 1]
```diff
@@ -0,0 +1,764 @@
...
+        flops = Ho * Wo * self.embed_dim * self.in_chans * (
+            self.patch_size[0] * self.patch_size[1])
+        if self.norm is not None:
+            flops += Ho * Wo * self.embed_dim
+        return flops
+
+
+class SwinTransformer(nn.Layer):
+    """ Swin Transformer
+        A PyTorch impl of : `Swin Transformer: Hierarchical Vision Transformer using Shifted Windows`  -
```
Comment (TingquanGao) [inline ppcls/modeling/architectures/swin_transformer.py]: PyTorch -> PaddlePaddle ？

Comment (littletomatodonkey) [reply ppcls/modeling/architectures/swin_transformer.py]: done，thx

**Review by TingquanGao — APPROVED** [review]

**Comment by zhangwenwen** [comment]
need to recompile paddle-paddle library manually?

**Comment by Darki-luo** [comment]
> att, four models top1 accs on ImageNet1k are as follows.
> 
> ```
> base_patch4_window12_384: {'top1_acc': 0.84390}
> base_patch4_window7_224: {'top1_acc': 0.8299600000762939}
> small_patch4_window7_224: {'top1_acc': 0.82746}
> tiny_patch4_window7_224: {'top1_acc': 0.806879999961853}
> ```
> 
> * todo
>   
>   * support training
> * Current problem
>   
>   * If you train the transformer using GPU and encountered the following problem, you can set `arch=sm_61`, more detailed info can be seen in [link](https://github.com/vchoutas/torch-mesh-isect/issues/6)
> 
> ```
> ERROR - (Fatal) Operator uniform_random raises an thrust::system::system_error exception.
> The exception content is
> :parallel_for failed: cudaErrorInvalidConfiguration: invalid configuration argument. (at /paddle/paddle/fluid/imperative/tracer.cc:172)
> ```

请问parallel_for failed的问题有跟进修复吗？

**Pull request merged**